qutip/core/qobj.py

```diff
@@ -96,18 +82,16 @@ def _require_equal_type(method):
     @functools.wraps(method)
     def out(self, other):
         if other == 0:
-            return method(self, other)
+            return self.copy()
```

Comment:
I don't think that is a good enough reason by itself to remove the call to `method` since it would break potential other users and nothing about the name of the decorator suggests it is not generally applicable.